pkg/services/accesscontrol/migrator/migrator.go

```diff
@@ -0,0 +1,44 @@
+package migrator
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func MigrateScopeSplit(db db.DB, log log.Logger) error {
+	t := time.Now()
+	var count = 0
+	err := db.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		var permissions []accesscontrol.Permission
+
+		err := sess.SQL("SELECT * FROM permission WHERE NOT scope = '' AND kind = '' AND attribute = '' AND identifier = ''").Find(&permissions)
+		if err != nil {
+			return err
+		}
+
+		for i, p := range permissions {
+			count++
+			kind, attribute, identifier := p.SplitScope()
+
+			permissions[i].Kind = kind
+			permissions[i].Attribute = attribute
+			permissions[i].Identifier = identifier
+
+			_, err := sess.Exec("UPDATE permission SET kind = ?, attribute = ?, identifier = ? WHERE id = ?", permissions[i].Kind, permissions[i].Attribute, permissions[i].Identifier, permissions[i].ID)
```

Comment:
Thought: instead of doing individual permission updates, we could delete the existing permissions and do a batch insert. It is slightly riskier (as it deletes the existing entries), but it should be faster, especially for large instances with lots of permissions. Let me know if you think it's worth doing.

Reply:
Good idea :) If you do the insert before the delete it's safer

Reply:
I wonder if we can do batching or concurrent batching in a transaction :thinking: